driver/pace/driver/driver.py

```diff
@@ -70,9 +70,13 @@ class DriverConfig:
     nz: int
     layout: Tuple[int, int]
     dt_atmos: float
-    diagnostics_config: diagnostics.DiagnosticsConfig
-    performance_config: PerformanceConfig
-    comm_config: CommConfig
+    diagnostics_config: diagnostics.DiagnosticsConfig = dataclasses.field(
+        default_factory=diagnostics.DiagnosticsConfig
+    )
+    performance_config: PerformanceConfig = dataclasses.field(
+        default_factory=PerformanceConfig
+    )
+    comm_config: CommConfig = dataclasses.field(default_factory=CommConfig)
```

Comment:
Is the idea of having default type for these to make demo simple? One argument is that not having default type ensures users to specify everything they're supposed to in the namelist.

Reply:
Not really - CommConfig and the others here already have defaults set for all of their options, so the idea was that if you can and would ever set `comm_config: {}` it's probably better to be able to remove this from the yaml entirely if you want the default.

If we wanted this to be required, we should make it so that there is no default communication type ("mpi" currently). But I feel like it makes sense as a default behavior.